optimum/intel/openvino/modeling_base.py

```diff
@@ -332,8 +332,12 @@ def _from_transformers(
     def compile(self):
         if self.request is None:
             logger.info("Compiling the model...")
-            cache_dir = Path(self.model_save_dir).joinpath("model_cache")
-            ov_config = {**self.ov_config, "CACHE_DIR": str(cache_dir)}
+            ov_config = {**self.ov_config}
```

Comment:
```suggestion
            ov_config = self.ov_config
```

Reply:
Agreed, also what do you think about modifying `self.ov_config` directly so that we don't do this for each compilation step ?

Reply:
@helena-intel  Do you agree to move this logic to __init__?

Reply:
Would definitely make sense to me, wdyt @helena-intel ?

Reply:
I have no preference on implementation, but IMO setting a config belongs to model compile, not model init (in OpenVINO the config is also set on compile_model, not read_model). So you should be able to do `model.compile(ov_config={"INFERENCE_PRECISION_HINT":"f32"})` and still get automatic model caching.

Reply:
@helena-intel This change (set default CACHE_DIR in __init__) doesn't impact the further implementation (if it will happen) to overwrite any ov_config with compile() API.

Reply:
It's not uncommon to want to change ov_config after loading the model, for example:

```python
model = OVModelFor....from_pretrained(model_id)
# do things
model.ov_config = {"INFERENCE_PRECISION_HINT": "f32"}
model.to("GPU")
model.compile()
```

For example in a gradio app where people can select different options, or in a benchmarking script where you want to iterate over several ov_config values.

With the new implementation this will have the side effect that there is no model caching anymore. Since the effect of caching is so large, and also since some people include model loading in "total inference time" (which is not good practice, but that's a separate discussion), it is important that we always enable caching by default, so I prefer the current behaviour where  model.compile() ensures that CACHE_DIR is set, with the addition that this should be modifyable. Maybe it can just be e.g. `self.ov_config.setdefault("CACHE_DIR", default_cache_dir)` in model.compile()?

Reply:
Agree. I don't aware that ov_config is a public member, I don't know why Python doesn't like private member. I changed it back to previous implementation. From my point of view, we should not change self.ov_config in compile.